### PR TITLE
[FCOS] Disable private imagestreams in samples operator

### PR DESCRIPTION
**Pull request by msheldyakov**
Fixes https://github.com/openshift/okd/issues/34

### CODE REVIEW

File: pkg/asset/templates/content/openshift/community-operators.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ const (
 
 var _ asset.WritableAsset = (*CommunityOperators)(nil)
 
-// CommunityOperators is the operatorHub setting for OKD to use community operators only
```
Comment (vrutkovs) [inline pkg/asset/templates/content/openshift/community-operators.go]: We shouldn't be touching that :)

Comment (msheldyakov) [reply pkg/asset/templates/content/openshift/community-operators.go]: oops. fixed. thanks!

**Comment by LorbusChris** [comment]
Thank you @msheldyakov!
yaml lint reports:
``` 
./data/data/manifests/openshift/samples-operator.yaml
  74:9      error    no new line character at the end of file  (new-line-at-end-of-file) 
```

**Comment by LorbusChris** [comment]
nvmd, that's fixed already!

**Comment by LorbusChris** [comment]
@msheldyakov could you squash all commits into one?

**Comment by msheldyakov** [comment]
yes, done

**Comment by LorbusChris** [comment]
@msheldyakov could you now just retitle the message and remove the `Squash: `? Unfortunately we can't use the GH UI to do that (neither can we do the squashing itself).

File: data/data/manifests/openshift/samples-operator.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+apiVersion: samples.operator.openshift.io/v1
+kind: Config
+metadata:
+  name: cluster
+spec:
+  architectures:
+    - x86_64
```
Comment (msheldyakov) [inline data/data/manifests/openshift/samples-operator.yaml]: without arch:
"'cannot change architectures from []string{"x86_64"} to []string(nil)'"

Comment (vrutkovs) [reply data/data/manifests/openshift/samples-operator.yaml]: Good catch. I suppose its a samples operator bug probably, but this solution is temporary anyway, lets roll with it

**Comment by vrutkovs** [comment]
Oh, we now fail "Failed to import expected imagestreams" test :)

/hold

until we sort this out

**Comment by vrutkovs** [comment]
Seems this needs to be gated by "redhat.io" check in pull secret

**Comment by msheldyakov** [comment]
> Seems this needs to be gated by "redhat.io" check in pull secret

do my best

File: pkg/asset/manifests/openshift.go [hunk 3]
```diff
@@ -206,6 +208,13 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 		"99_community_operators.yaml":       applyTemplateData(communityOperatorsSetting.Files()[0].Data, templateData),
 	}
 
+	// Private images from redhat.io are not available on fcos
+	if !strings.Contains(installConfig.Config.PullSecret, "redhat.io"){
```
Comment (msheldyakov) [inline pkg/asset/manifests/openshift.go]: > Seems this needs to be gated by "redhat.io" check in pull secret

**Comment by LorbusChris** [comment]
@msheldyakov gofmt isn't happy yet :)

**Comment by vrutkovs** [comment]
/retest

Masters didn't come up

**Comment by vrutkovs** [comment]
/retest

Still no luck bringing master up

**Comment by LorbusChris** [comment]
the sample operator errors definitely look related to this change

**Comment by vrutkovs** [comment]
>the sample operator errors definitely look related to this change

yeah, so we're trying out an option to create this setting of pull-secret doesn't have "redhat.io" auth. Probably it would best be solved on samples-operator side though?

File: data/data/manifests/openshift/samples-operator.yaml [hunk 4]
```diff
@@ -0,0 +1,321 @@
+# yamllint disable rule:document-start
```
Comment (msheldyakov) [inline data/data/manifests/openshift/samples-operator.yaml:1]: An alternative is to describe each imagestream in a separate file?

**Comment by msheldyakov** [comment]
"Message: "imagestreams.image.openshift.io \"php\" not found""

**Comment by msheldyakov** [comment]
image.openshift.io/v1 is not available while applying this manifest. Bootstrap is never successful.

I will try to fix it on the samples-operator side. Gist for temporary manual fixing after install on okd:
https://gist.github.com/msheldyakov/a8c3ab7176328736dae4d834f38b92e6

**Pull request closed without merging**